plugin-modernizer-core/src/test/java/io/jenkins/tools/pluginmodernizer/core/recipes/DeclarativeRecipesTest.java

```diff
@@ -612,8 +612,9 @@ void shouldRemoveExtraProperties() {
     void upgradeToRecommendCoreVersionTest() {
         rewriteRun(
                 spec -> spec.recipeFromResource(
-                        "/META-INF/rewrite/recipes.yml",
-                        "io.jenkins.tools.pluginmodernizer.UpgradeToRecommendCoreVersion"),
+                                "/META-INF/rewrite/recipes.yml",
+                                "io.jenkins.tools.pluginmodernizer.UpgradeToRecommendCoreVersion")
+                        .expectedCyclesThatMakeChanges(2),
```

Comment:
Nope. This is not expected. One cycle must be enough.

This indicate a bug in the recipe since it's not stable after 1 cycle

Reply:
I saw some occurrence of multiple cycle on recipe order. Sometimes it's about direct visitor invocation vs doAfterVisit.

Sometimes it's about direct XML manipulation vs Maven manipulation (maybeRefreshModel?)

Only solution is by checking logs and taking debugger to see the difference between first invocation and second.